PropertyTax.java

```
@@ -0,0 +1,110 @@
+public class PropertyTax {
+
+  private final double ASSESSMENT_MODIFIER = 0.6;
+  private final double RESIDENTIAL_RATE = 0.821;
+  private final double COMMERCIAL_RATE = 1.78;
+  private final double FARM_RATE = 0.152;
+  private final double MUNICIPAL_SHARE = 0.471;
+  private final double REGIONAL_SHARE = 0.332;
+  private final double EDUCATION_SHARE = 1 - MUNICIPAL_SHARE - REGIONAL_SHARE;
+  private final String RESIDENTIAL = "residential";
+  private final String COMMERCIAL = "commercial";
+  private final String FARM = "farm";
+  private double propertyValue;
+  private String propertyType;
+
+  public PropertyTax() {}
+
+  public PropertyTax(char propertyInput, double valueInput) {
+
+    switch (Character.toLowerCase(propertyInput)) {
+      case 'r':
+        propertyType = RESIDENTIAL;
+        break;
+      case 'c':
+        propertyType = COMMERCIAL;
+        break;
+      case 'f':
+        propertyType = FARM;
+        break;
+    }
+    propertyValue = valueInput;
+  }
+
+  public String getPropertyType() {
+    return propertyType;
+  }
+
+  public void setPropertyType(char newProperty) {
+    switch (Character.toLowerCase(newProperty)) {
```

can you add a default case to handle errors? so if someone enters "x" then it would throw an error

---

I should have but I was lazy and we haven't learned exception handling for Java yet. In Python I'd raise an error.